src/hotspot/cpu/x86/templateTable_x86.cpp

```diff
@@ -4048,11 +4048,17 @@ void TemplateTable::_new() {
   __ load_resolved_klass_at_index(rcx, rcx, rdx);
   __ push(rcx);  // save the contexts of klass for initializing the header
 
-  // make sure klass is initialized & doesn't have finalizer
-  // make sure klass is fully initialized
-  __ cmpb(Address(rcx, InstanceKlass::init_state_offset()), InstanceKlass::fully_initialized);
-  __ jcc(Assembler::notEqual, slow_case);
+  // make sure klass is initialized
+  if (VM_Version::supports_fast_class_init_checks()) {
```

Comment:
Maybe this should use conditional compilation and an assert rather than a dynamic runtime check, as we expect this to always, and only, be true on 64-bit.